services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/northbound/switches/SwitchFailuresSpec.groovy

```diff
@@ -134,4 +140,55 @@ class SwitchFailuresSpec extends BaseSpecification {
         and: "Remove the flow"
         flowHelper.deleteFlow(flow.id)
     }
+
+    @Ignore("This is a known Kilda limitation and feature is not implemented yet.")
+    def "System can handle situation when switch reconnects while flow is being rerouted"() {
+        given: "A flow with alternative paths available"
+        def switches = topology.getActiveSwitches()
+        List<List<PathNode>> allPaths = []
+        def (Switch srcSwitch, Switch dstSwitch) = [switches, switches].combinations()
+                .findAll { src, dst -> src != dst }.unique { it.sort() }.find { Switch src, Switch dst ->
+            allPaths = db.getPaths(src.dpId, dst.dpId)*.path
+            allPaths.size() > 1
+        } ?: assumeTrue("No suiting switches found", false)
+        def flow = flowHelper.randomFlow(srcSwitch, dstSwitch)
+        flowHelper.addFlow(flow)
+        def currentPath = PathHelper.convert(northboundService.getFlowPath(flow.id))
+
+        and: "There is a more preferable alternative path"
+        def alternativePaths = allPaths.findAll { it != currentPath }
+        def preferredPath = alternativePaths.first()
+        def uniqueSwitch = pathHelper.getInvolvedSwitches(preferredPath).find {
+            !pathHelper.getInvolvedSwitches(currentPath).contains(it)
+        }
```

Comment:
I would rather add assumeTrue() here with some message instead of asserting.

Reply:
agree